test/test_autograd.py

```diff
@@ -2142,6 +2142,42 @@ def backward(ctx, grad_output):
         out.sum().backward()
         self.assertEqual(x.grad, y_data)
 
+    def test_reentrant_child_error(self):
+        class BackwardError(Function):
+            @staticmethod
+            def forward(ctx, inp):
+                return inp
+
+            @staticmethod
+            def backward(ctx, grad):
+                raise Exception('simulate error')
+
+        # Parent graph.
+        a = torch.rand(3, 3, requires_grad=True)
+        b = torch.rand(3, 3, requires_grad=True)
```

Comment:
nit: maybe declare `b` below to make it obvious that the two graphs are independent.